src/memberAPI.ts

```diff
@@ -1,70 +1,217 @@
-          return;
+          return {
+            error: "Couldn't edit user with undefined email!",
+            status: 400,
+          };
         }
-        db.doc('members/' + req.body.email).set(req.body).then(() => {
-          res.status(200).json({ status: "Success", member: req.body });
-        }).catch((reason) => {
-          res.status(200).json({ error: "Couldn't edit user for reason: " + reason });
-        });
+        db.doc('members/' + req.body.email)
+          .set(req.body)
+          .then(() => {
+            return { status: 200, member: req.body };
+          })
+          .catch((reason) => {
+            return {
+              error: "Couldn't edit user for reason: " + reason,
+              status: 500,
+            };
+          });
       }
     }
   }
 };
 
-export let deleteMember = async (req, res) => {
+export let updateMember = async (
+  req: Request,
+  res: Response
+): Promise<MemberResponse | ErrorResponse> => {
   if (checkLoggedIn(req, res)) {
-    let member = await (await db.doc('members/' + req.session.email).get()).data();
+    let member = await (
+      await db.doc('members/' + req.session.email).get()
+    ).data();
     if (!member) {
-      res.status(200).json({ error: "Not member with email: " + req.session.email });
+      return {
+        error: 'No member with email: ' + req.session.email,
+        status: 401,
+      };
+    } else {
+      let canEdit: boolean = PermissionsManager.canEditMembers(member.role);
+      if (!canEdit && member.email !== req.body.email) {
+        return {
+          error:
+            'User with email: ' +
+            req.session.email +
+            ' does not have permission to edit members!',
+          status: 403,
+        };
+      } else {
+        if (!req.body.email || req.body.email === '') {
+          return {
+            error: "Couldn't edit user with undefined email!",
+            status: 400,
+          };
+        }
+        if (
+          (req.body.role || req.body.first_name || req.body.last_name) &&
+          !canEdit
+        ) {
+          return {
+            status: 403,
+            error:
+              'User with email: ' +
+              req.session.email +
+              ' does not have permission to edit member name or roles!',
+          };
+        }
+        let response: ErrorResponse | MemberResponse = await db
+          .doc('members/' + req.body.email)
+          .update(req.body)
+          .then(() => {
+            return {
+              member: req.body as Member,
+              status: 200,
+            };
+          })
+          .catch((reason) => {
+            return {
+              error: "Couldn't edit user for reason: " + reason,
+              status: 500,
+            };
+          });
+
+        return response;
+      }
+    }
+  }
+};
+
+export let getMember = async (
+  req: Request,
+  res: Response
+): Promise<MemberResponse | ErrorResponse> => {
+  console.log(req.session.email);
```

Comment:
I think this line can be removed or commented out.

Reply:
Thanks for the feedback I've updated those three lines.